libqtile/bar.py

```diff
@@ -160,7 +160,7 @@ class Bar(Gap, configurable.Configurable):
         ("background", "#000000", "Background colour."),
         ("opacity", 1, "Bar window opacity."),
         ("margin", 0, "Space around bar as int or list of ints [N E S W]."),
-        ("depth", 32, "Desired colour depth for bar")
+        ("depth", 24, "Desired colour depth for bar. 32 bit depth needed for transparency.")
```

Comment:
I've defaulted to 24 bit as the default background is "#000000".

We could check if background has requested transparency and flip this to "32" when the bar is initialised. Any objections/concerns?

Reply:
Is there any reason to have this configurable at all? We can default to 32 in all cases (don't bother checking background colours - the underlying window doesn't know what's going on it), only falling back to 24 if for whatever reason the system/X server doesn't support 32 bit. Saying that, I don't think implementing the fallback makes sense until someone reports it is needed -- maybe everyone running X these days can use 32 bit windows. So maybe hard-code 32bits everywhere?

Reply:
I did think about that and I agree it's probably the right answer but I was taking a cautious approach to avoid any breakages.

As for removing fallbacks, we can keep this bit in and just log a message if there isn't a 32 bit visual available:
https://github.com/qtile/qtile/blob/9f513d157dcd4edac49cdc56ad2260eb28e42406/libqtile/backend/x11/xcbq.py#L315-L322

Reply:
Thinking about it some more, maybe a better approach is to set a `default_depth` and `default_visual` and have these be 32 bit unless it's not available and then it falls back to `root_depth` and the equivalent visual.

That should probably help remove some code.

Reply:
Whatever you think will be the nicest!